### PR TITLE
hwmv2: convert NXP S32 SoC and boards to hardware model v2

### PR DESCRIPTION
**Pull request by manuargue**
Convert NXP S32 SoC and boards to hardware model v2. Depends on [hal_nxp#340.](https://github.com/zephyrproject-rtos/hal_nxp/pull/340)

### CODE REVIEW

File: soc/nxp/s32/s32k1/Kconfig.soc [hunk 1]
```diff
@@ -1,65 +1,56 @@
...
 
-choice
-	prompt "NXP S32K1XX MCU selection"
-	depends on SOC_SERIES_S32K1XX
+config SOC_SERIES_S32K1XX
+	bool
+	select SOC_FAMILY_NXP_S32
+
+config SOC_SERIES
+	default "s32k1" if SOC_SERIES_S32K1XX
```
Comment (nordicjm) [inline soc/nxp/s32/s32k1/Kconfig.soc:11]:
```suggestion
	default "s32k1xx" if SOC_SERIES_S32K1XX
```

File: soc/nxp/s32/s32k3/Kconfig.soc [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+# Copyright 2023-2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+config SOC_SERIES_S32K3XX
+	bool
+	select SOC_FAMILY_NXP_S32
+
+config SOC_SERIES
+	default "s32k3" if SOC_SERIES_S32K3XX
```
Comment (nordicjm) [inline soc/nxp/s32/s32k3/Kconfig.soc:11]:
```suggestion
	default "s32k3xx" if SOC_SERIES_S32K3XX
```

File: soc/nxp/s32/soc.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+family:
+- name: nxp_s32
+  series:
+  - name: s32k1
```
Comment (nordicjm) [inline soc/nxp/s32/soc.yml:4]:
```suggestion
  - name: s32k1xx
```

File: soc/nxp/s32/soc.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    socs:
+    - name: s32k116
+    - name: s32k118
+    - name: s32k142
+    - name: s32k142w
+    - name: s32k144
+    - name: s32k144w
+    - name: s32k146
+    - name: s32k148
+  - name: s32k3
```
Comment (nordicjm) [inline soc/nxp/s32/soc.yml:14]:
```suggestion
  - name: s32k3xx
```

File: soc/nxp/s32/s32k1/Kconfig.soc (hunk 1 above)
Comment (nordicjm) [reply soc/nxp/s32/s32k1/Kconfig.soc:11]: Thanks, have created an issue: https://github.com/zephyrproject-rtos/zephyr/issues/69317

File: soc/nxp/s32/Kconfig.defconfig [hunk 5]
```diff
@@ -0,0 +1,8 @@
+# Copyright 2022,2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+if SOC_FAMILY_NXP_S32
+
+rsource "*/Kconfig.defconfig"
+
+endif # SOC_FAMILY_KINETIS
```
Comment (iuliana-prodan) [inline soc/nxp/s32/Kconfig.defconfig:8]: s/SOC_FAMILY_KINETIS/SOC_FAMILY_NXP_S32

**Review by nordicjm — APPROVED** [review]

**Review by bperseghetti — APPROVED** [review]

**Review by iuliana-prodan — APPROVED** [review]

**Pull request merged**